blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineNodeUtil.java

```diff
@@ -59,19 +66,68 @@ public static NodeRunStatus getStatus(WorkflowRun run){
     }
 
 
-    public static String getDisplayName(FlowNode node) {
+    @Nonnull
+    public static String getDisplayName(@Nonnull FlowNode node) {
         return node.getAction(ThreadNameAction.class) != null
             ? node.getAction(ThreadNameAction.class).getThreadName()
             : node.getDisplayName();
     }
 
     public static boolean isStage(FlowNode node){
-        return node !=null && (node.getAction(StageAction.class) != null
+        return node !=null && ((node.getAction(StageAction.class) != null && !isSyntheticStage(node))
             || (node.getAction(LabelAction.class) != null && node.getAction(ThreadNameAction.class) == null));
 
     }
 
-    public static boolean isParallelBranch(FlowNode node){
+    public static boolean isSyntheticStage(@Nullable FlowNode node){
+        return node!= null && getSyntheticStage(node) != null;
+    }
+
+    @CheckForNull
+    public static TagsAction getSyntheticStage(@Nullable FlowNode node){
+        if(node != null) {
+            for (Action action : node.getActions()) {
+                if (action instanceof TagsAction && ((TagsAction) action).getTagValue("SYNTHETIC_STAGE") != null) {
```

Comment:
I take that back - `ExtensionList.lookup(StageTagsMetadata.class).get(SyntheticStage.class)` does the trick, no new code needed.